godotproject/Player.gd

```diff
@@ -53,10 +69,7 @@ func start_swing():
 	var swing_vec = update_swing_radius_angle()
 	var swing_dir_vec = swing_vec.rotated(PI/2)
 	var dotprod = swing_dir_vec.dot(velocity)
-	if abs(dotprod)<  0.01 or swing_radius < 0.01:
-		swing_angular_speed = 0
-	else:
-		swing_angular_speed = dotprod / abs(dotprod) * pow(abs(dotprod), 0.5) / max(1, swing_radius)
+	swing_angular_speed = sign(dotprod) * pow(abs(dotprod), 0.5) / max(1, swing_radius)
```

Comment:
Using the sign(x) function instead of x / abs(x) really helps condense this code down, and eliminates the need for a special case when dotprod == 0